### ISSUE TITLE
[Accessibility]: Search box ARC checklist issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34733]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

- [ ] SVG icons inside of toggles should be `aria-hidden`
- [ ] Search box Monaco editor should have `role="textbox"` (currently has `role="code"` which is invalid)

### Expected behavior

Issues fixed and no longer flagged by ARC toolkit

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
